### ISSUE TITLE
[Bug / Feature]: Update Highlights for Negative Binomial

### ISSUE THREAD

**Issue by levolz** [jasp-stats/jasp-issues#1663]
### JASP Version

0.16.1

### JASP Module

jaspDistributions

### What analysis are you seeing the problem on?

Negative Binomial

### What OS are you seeing the problem on?

Windows 11 (& others)

### Bug Description

Highlighting the CDF of an interval includes the lower bound, i.e. difference between both cdf highlights does not equal the cumulative probability over the interval.

### Expected Behaviour

The lower CDF highlight should give the [0, a -1] cdf instead of [0, a] cdf.

### Steps to Reproduce

Highlighting CDF in the negative binomial analysis in the cumulative distribution plot

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if any.
- [X] I have included a JASP file that causes the crash/bug, if any.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by Kucharssim** [comment]
Thanks for filing the report!

> difference between both cdf highlights does not equal the cumulative probability over the interval.

I think the problem is that there are no two cdf highlights. There are two probability masses shown, and one cumulative probability shown. The two values at the end of the "highlights" merely show the probability mass of the value of the random variable, not the cdf. The value associated with the "cumulative probability" concerns the whole interval that is colored, and specified according to the interval in the options (which indeed includes both the lower and upper bounds)
![Screenshot 2022-03-17 at 17 19 23](https://user-images.githubusercontent.com/7093546/158845950-70c81f18-88db-4743-ab7b-ca5f90970471.png)

In other words, the value of the cumulative probability is the sum of the probability masses of all values within the specified interval, not only the lower and upper bounds.

Does that clarify the issue?

I see that the options/plots are somewhat confusing for the discrete distributions. The concept of the plot borrows from the plots in the continuous distributions, where the distinction between density at a specified point vs. the probability mass inside of the interval is easier to spot, whereas for the discrete distributions it is perhaps confusing to understand what part of the plot relates to the cdf and what to the pmf. I would be happy to hear suggestions on how to clarify/change this.

**Comment by levolz** [comment]
Thanks for the explanation!
The PDF values within the discrete probability plots are indeed a bit
confusing :D

However, I mean that the CDF highlight in the PDF plot should be the
difference between the CDF highlights in the CDF plot (i.e., the issue
concerns the CDF plot).
Hope the picture below illustrates what I mean better than the words -
i.e., if below you'd show the CDF(a - 1) = 0.19, they'd add up, while
currently, the CDF(a) does not really add informational value to the
analysis.

[image: grafik.png]

**Comment by levolz** [comment]
CDF(a-1) = 0.11, ofc. - my mistake

**Comment by Kucharssim** [comment]
Not sure why my previous reply is deleted...

> (i.e., the issue concerns the CDF plot).

You are right. I was only looking at the PDF plot.

**Comment by levolz** [comment]
No, I think the issue persists (if I am not mistaken) - see screenshot:
![image](https://github.com/jasp-stats/jasp-issues/assets/42385806/8e7a79af-a1bd-4fbe-9851-3846a3023cd7)

The issue was referring to CDF(2 <= x <= 5) = CDF(x <= 5) - CDF(x < 2) ~= .98 - .75 = .23 --- however, the highlighted CDF values are .98 (upper) and .88 (lower), which conceptually does not make sense - it should instead be at the lower end of the highlighted PMF(x = 2) = .13, corresponding to CDF(x < 2) = .75

**Comment by tomtomme** [comment]
Thx for the details. I can confirm this bug still exists also on my end. I think JASP outputting reslults that are not conform with the users input warrants a high prio for this bug. Lets hope the "high" label will help fixing it faster.

**Comment by tomtomme** [comment]
File to test:
[d.jasp.zip](https://github.com/user-attachments/files/18053786/d.jasp.zip)

**Comment by tomtomme** [comment]
@Kucharssim 
any chance this very old error could be fixed for 0.95?
